### DISCUSSION TITLE
When I try to play multiple files, the app crashes.

### DISCUSSION THREAD

**Discussion by kabiri** [MagicFoundation/Alcinoe#291] [Q&A]
Hi dear @Zeus64 
I have multiple files and I play them one by one. But when I try to play some of them, the player shows an error status.
After playing several files, the application closes once.

my code is :
```
Procedure Play(fileN : string)
begin
 if ALVideoPlayerSurface1.VideoPlayer.state in [vpsStarted] then
 begin
   ALVideoPlayerSurface1.VideoPlayer.Stop;
   if not (ALVideoPlayerSurface1.VideoPlayer.state in [vpsIdle]) then
     ALVideoPlayerSurface1.resetVideoPlayer;
 end;
 if ALVideoPlayerSurface1.VideoPlayer.state in [vpsError] then
 begin
     ALVideoPlayerSurface1.resetVideoPlayer;
 end;
 if ALVideoPlayerSurface1.VideoPlayer.state in [vpsIdle] then
 begin
   ALVideoPlayerSurface1.VideoPlayer.setLooping(true);
   ALVideoPlayerSurface1.VideoPlayer.prepare(fileN, True{AutoStartWhenPrepared});
 end;
end;
```
I'm testing on Android.
Files that cause errors on Android play without any problems on Windows.
How can I figure out where the problem is coming from?
How can I run and test the program in Windows?

------EDIT-------
Your test app shows the video, but when I add its link to my program, it doesn't play and the component goes to the vpsError state.

**Comment by Zeus64** [comment]
can you try to see the log in the android monitor tool ?

**Reply by kabiri** [reply]
I don't see any error in the log. What should I say about the log?
Do you need me to email it to you?

**Comment by Zeus64** [comment]
are you sur you use correctly the android monitor ? mine look like this :

![image](https://github.com/MagicFoundation/Alcinoe/assets/4325047/a7d36455-8253-405e-adbe-3551df562fb7)

**Reply by kabiri** [reply]
I download android viewer from here : [http://app.serialgames.jp/sgd/alog/download/androidlogviewer_win_2_0_0.zip](url)

**Comment by kabiri** [comment]
I use this :
![image](https://github.com/MagicFoundation/Alcinoe/assets/8082245/337a7bf8-d396-4444-9e7c-2aca07d86391)
Since android monitor is not included with Delphi 11.

**Reply by kabiri** [reply]
Maybe it's here. The program closed after these.

05-10 23:30:33.211 10362 10362 E AndroidRuntime FATAL EXCEPTION: main
05-10 23:30:33.211 10362 10362 E AndroidRuntime Process: com.embarcadero.PlayerTest, PID: 10362
05-10 23:30:33.211 10362 10362 E AndroidRuntime com.embarcadero.rtl.NativeDispatchException: Invoke error: method 'onFrameAvailable(Landroid/graphics/SurfaceTexture;)V' not found
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at com.embarcadero.rtl.ProxyInterface.dispatchToNative2(Native Method)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at com.embarcadero.rtl.ProxyInterface.invoke(ProxyInterface.java:20)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at $Proxy14.onFrameAvailable(Unknown Source)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at android.graphics.SurfaceTexture$1.handleMessage(SurfaceTexture.java:214)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at android.os.Handler.dispatchMessage(Handler.java:106)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at android.os.Looper.loopOnce(Looper.java:226)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at android.os.Looper.loop(Looper.java:313)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at android.app.ActivityThread.main(ActivityThread.java:8757)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at java.lang.reflect.Method.invoke(Native Method)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
05-10 23:30:33.211 10362 10362 E AndroidRuntime 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

**Comment by kabiri** [comment]
Do you have any suggestion?
@Zeus64

**Comment by kabiri** [comment]
Good day,

Did this information help in identifying the issue?
In your opinion, is the problem with the component or the Android player?
Thanks

[playerError.log](https://github.com/MagicFoundation/Alcinoe/files/11524178/playerError.log)
or
https://mega.nz/file/tlwkiRDA#gHQ1pBauz0DCH2pOKIb7w11hoVC_QdYtE1UpbQm7OrY

**Reply by Zeus64** [reply]
Probably the file you try to play is not supported :(

**Reply by kabiri** [reply]
> Probably the file you try to play is not supported :(

No, sometimes it displays it without any issues.
Sometimes it crashes and closes the application, and other times an error event related to the player occurs.

**Reply by kabiri** [reply]
In this video, it is clear that the file is supported properly. I ran the program on two phones, Samsung A32 on the right side and A7 on the left side. You can see the functionality of the program (also, the file named "11" is the one you uploaded).

[https://mega.nz/file/5wY0kS5R#maLuEV8gQDVEpKG68DzF0AypMYzlvhRbTgbRLHe8LOM](url)

**Comment by kabiri** [comment]
@Zeus64 
Good day!
Have you identified the problem and have you investigated it?
I'm sure the files are supported for playback because sometimes they are displayed and sometimes they are not.
But even if they are not supported, why does the program close?
What should I do to prevent the program from closing?
